doc/api/esm.md

```diff
@@ -175,89 +175,88 @@ unspecified.
-`"exports"`. [Conditional Exports][] can also be used within `"exports"` to
-define different package entry points per environment.
 
-#### `package.json` `"main"`
+The `"exports"` field provides an alternative to `"main"` where the package
+main entry point can be defined while also encapsulating the package, preventing
+any other entry points besides those defined in `"exports"`. If package entry
+points are defined in both `"main"` and `"exports"`, the latter takes precedence
+in versions of Node.js that support `"exports"`. [Conditional Exports][] can
+also be used within `"exports"` to define different package entry points per
+environment.
 
-The `package.json` `"main"` field defines the entry point for a package,
-whether the package is included into CommonJS via `require` or into an ES
-module via `import`.
+If both `"exports"` and `"main"` are defined, the `"exports"` field takes
+precedence over `"main"`.
+
+All package entry points apply to both `import` and `require`; `"exports"` and
+`"main"` are **not** specific to ES modules or CommonJS.
+
+This is important with regard to `require`, since `require` of ES module files
+throws an error in all versions of Node.js. To create a package that works both
+in modern Node.js via `import` and `require` and also legacy Node.js versions,
+see [the dual CommonJS/ES module packages section][].
+
+#### Package Exports Main
+
+To set the main entry point for a package, it is advisable to use both
+`"exports"` and the `"main"`:
 
 <!-- eslint-skip -->
 ```js
-// ./node_modules/es-module-package/package.json
 {
-  "type": "module",
-  "main": "./src/index.js"
+  "main": "./main.js",
+  "exports": "./main.js"
 }
 ```
 
-```js
-// ./my-app.mjs
-
-import { something } from 'es-module-package';
-// Loads from ./node_modules/es-module-package/src/index.js
-```
-
-An attempt to `require` the above `es-module-package` would attempt to load
-`./node_modules/es-module-package/src/index.js` as CommonJS, which would throw
-an error as Node.js would not be able to parse the `export` statement in
-CommonJS.
+The benefit of doing this is that when using the `"exports"` field all
+subpaths of the package will no longer be available to importers under
+`require('pkg/subpath.js')`, and instead they will get a new error,
+`ERR_PACKAGE_PATH_NOT_EXPORTED`.
 
-As with `import` statements, for ES module usage the value of `"main"` must be
-a full path including extension: `"./index.mjs"`, not `"./index"`.
+This new encapsulation of exports provides new more reliable guarantees
+about package interfaces for tools and when handling semver upgrades for a
+package. It is not a strong encapsulation since a direct require of any
+absolute subpath of the package such as
+`require('/path/to/node_modules/pkg/subpath.js')` will still work correctly.
 
-If the `package.json` `"type"` field is omitted, a `.js` file in `"main"` will
-be interpreted as CommonJS.
+#### Package Exports Subpaths
```

Comment:
```suggestion
#### Package Subpaths
```

“Exports” here is being used to refer to `"exports"`, e.g. “Package `"exports"` Subpaths”, but that only makes sense if you already know what `"exports"` is. I think “Package Subpaths” is clearer as it’s generic.

Reply:
Subpaths exists as a concept without exports though.

We need to be clear that the exports subpaths are their own thing, specially since they define a new virtual specifier space for bare specifiers.

How about `Subpath Exports`?